pyperf/_utils.py

```diff
@@ -317,9 +320,33 @@ def __exit__(self, *args):
 class ReadPipe(_Pipe):
     def open_text(self):
         file = open(self._fd, "r", encoding="utf8")
+        if self._timeout:
+            os.set_blocking(file.fileno(), False)
         self._file = file
         return file
 
+    def read_text(self):
```

Comment:
Can you move the timeout parameter from `__init__()` to this method instead?

It's a little bit strange to pass a timeout to the constructor whereas it creates two file descriptors (read end and write end).

Reply:
Done.